www/SQLitePlugin.js

```diff
@@ -221,33 +225,46 @@
               
             };
         };
-      })(this);
-      openerrorcb = (function(_this) {
-        return function() {
+      openerrorcb = function() {
+          failed = true;
           console.log('OPEN database: ' + _this.dbname + ' FAILED, aborting any pending transactions');
           if (!!error) {
             error(newSQLError('Could not open database'));
           }
           delete _this.openDBs[_this.dbname];
           _this.abortAllPendingTransactions();
         };
-      })(this);
       this.openDBs[this.dbname] = { 
         state: DB_STATE_INIT,
         connections: []
       };
 
-      var options = {};
-      for (var key in this.openargs) {
-        options[key] = this.openargs[key];
+      var copyOptions = function() {
+        var options = {};
+        var optionNames = Object.keys(_this.openargs);
+        for (var i = 0; i < optionNames.length; i++) {
+          options[optionNames[i]] = _this.openargs[optionNames[i]];
+        }
+        return options;
       }
 
-      for (var i = 0; i < maxConnections; i++) {
-        var connectionName = 'connection' + i;
+      var openNext = function openNext(connNumber) {
+
+        var options = copyOptions();
+
+        var connectionName = 'connection' + connNumber;
+        var onSuccess = opensuccesscb(connectionName);
         options['connectionName'] = connectionName;
 
-        cordova.exec(opensuccesscb(connectionName), openerrorcb, "SQLitePlugin", "open", [options]);
-      }
+        cordova.exec(function() {
+          onSuccess();
+          if (connNumber < maxConnections) {
```

Comment:
Aren't we going to open 1 more connection than intended? We are only incrementing the connNumber variable after the check and we are starting on 0.